src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -504,6 +516,51 @@ bool ParserTableFunctionView::parseImpl(Pos & pos, ASTPtr & node, Expected & exp
     return true;
 }
 
+bool ParserFilterClause::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    assert(node);
+    ASTFunction & function = dynamic_cast<ASTFunction &>(*node);
+
```

Comment:
Maybe lets make argument of type `ASTFunction` to be explicit for caller and have less change of passing wrong input into `parseImpl`

Reply:
`parseImpl` is declared in `IParserBase` and is called by `IParserBase::parse`, so we hardly can change its signature as long is it remains to be `IParserBase`'s descendant.